assets/src/scripts/components/map.js

```diff
@@ -1,26 +1,122 @@
+const { select } = require('d3-selection');
+const { createStructuredSelector } = require('reselect');
+
 const { map: createMap, marker: createMarker } = require('leaflet');
-const { BasemapLayer, TiledMapLayer } = require('esri-leaflet');
+const { BasemapLayer, TiledMapLayer, dynamicMapLayer, Util } = require('esri-leaflet');
 
+const { link, provide } = require('../lib/redux');
 
-const HYDRO_URL = 'https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/Esri_Hydro_Reference_Overlay/MapServer';
+const { FLOOD_EXTENTS_ENDPOINT, FLOOD_BREACH_ENDPOINT, FLOOD_LEVEE_ENDPOINT } = require('../floodData');
+const { Actions } = require('../store');
 
+const FIM_ENDPOINT = window.FIM_ENDPOINT;
 
-function attachToNode(node, {latitude, longitude, zoom}) {
+const getLayerDefs = function(layerNo, siteno, stage) {
+   const stageQuery = stage ? ` AND STAGE = ${stage}` : '';
+   return `${layerNo}: USGSID = '${siteno}'${stageQuery}`;
+};
+/*
+ * Creates a site map
+ */
+const siteMap = function(node, {siteno, latitude, longitude, zoom}) {
     // Create map on node
-    const map = createMap(node, {
+    node.append('div')
+        .attr('id', 'site-map');
+    const map = createMap('site-map', {
         center: [latitude, longitude],
         zoom: zoom
     });
 
+    let floodLayer = dynamicMapLayer({
+        url: FLOOD_EXTENTS_ENDPOINT,
+        layers: [0],
+        f: 'image',
+        format: 'png8'
+    });
+    let breachLayer = dynamicMapLayer({
+        url: FLOOD_BREACH_ENDPOINT,
+        layers: [0],
+        f: 'image',
+        format: 'png8'
+    });
+    let leveeLayer = dynamicMapLayer({
+        url: FLOOD_LEVEE_ENDPOINT,
+        layers: [0, 1],
+        f: 'image',
+        format: 'png8',
+        layerDefs: `${getLayerDefs(0, siteno)};${getLayerDefs(1, siteno)}`
+    });
+
+    const updateFloodLayers = function (node, {stages, gageHeight}) {
+        if (gageHeight) {
+            const layerDefs = getLayerDefs(0, siteno, gageHeight);
+            floodLayer.setLayerDefs(layerDefs);
+            breachLayer.setLayerDefs(layerDefs);
+        }
+        if (stages.length === 0) {
+            if (map.hasLayer(floodLayer)) {
+                map.removeLayer(floodLayer);
+                map.removeLayer(breachLayer);
+                map.removeLayer(leveeLayer);
+            }
+        } else {
+            if (!map.hasLayer(floodLayer)) {
+                map.addLayer(floodLayer);
+                map.addLayer(breachLayer);
+                map.addLayer(leveeLayer);
+            }
+        }
+    };
+
+    const updateMapExtent = function (node, {extent}) {
+        if (Object.keys(extent).length > 0) {
+            map.fitBounds(Util.extentToBounds(extent));
+        }
+    };
+
     // Add a gray basemap layer
     map.addLayer(new BasemapLayer('Gray'));
 
     // Add the ESRI World Hydro Reference Overlay
-    map.addLayer(new TiledMapLayer({url: HYDRO_URL}));
+    if (window.HYDRO_URL) {
```

Comment:
Was this intended to be `window.HYDRO_ENDPOINT`? Is there a scenario we'd actually need this to be configurable? If not it seems unnecessary to make the Python-side aware of it.

Other thoughts... If we need to, we could consider using build-time variables for this sort of thing. `envify` is one way: https://github.com/hughsk/envify Perhaps if we need to share values between the two we could put them in JSON, and load it from config.py.

Reply:
It should be HYDRO_ENDPOINT. I did this for a couple of reasons. It would be good to put the configuration in one place and I wanted to be able to disable it for testing so that it doesn't create the layer (creating the layer would fetch the image then).

Reply:
I'll look into envify, but I was following the pattern that we already have since many of our endpoints are only used by the javascript.